--- /dev/null	2024-07-23 05:45:45.911999592 +0000
+++ b/package/min_prog_trace/Config.in	2024-07-23 06:33:23.575842947 +0000
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_MIN_PROG_TRACE
+    bool "min_prog_trace"
+    help
+      min_prog_trace is a minimal program for tracing.
+
+      https://github.com/alessandrocarminati/min_prog_trace
--- /dev/null	2024-07-23 05:45:45.911999592 +0000
+++ b/package/min_prog_trace/min_prog_trace.mk	2024-07-23 08:53:19.307994387 +0000
@@ -0,0 +1,15 @@
+MIN_PROG_TRACE_VERSION = e6a7def49ce6264ff4ed021d3363c04f15cfb7ca
+MIN_PROG_TRACE_SITE = https://github.com/alessandrocarminati/min_prog_trace
+MIN_PROG_TRACE_SITE_METHOD = git
+
+define MIN_PROG_TRACE_BUILD_CMDS
+    $(TARGET_MAKE_ENV) $(MAKE) -C $(@D) CROSS_COMPILE=$(TARGET_CROSS)
+endef
+
+define MIN_PROG_TRACE_INSTALL_TARGET_CMDS
+    $(INSTALL) -D -m 0755 $(@D)/min $(TARGET_DIR)/usr/bin/min
+    $(INSTALL) -D -m 0755 $(@D)/min.large $(TARGET_DIR)/usr/bin/min.large
+    $(INSTALL) -D -m 0755 $(@D)/ftrace_it $(TARGET_DIR)/usr/bin/ftrace_it
+endef
+
+$(eval $(generic-package))
--- a/package2/Config.in	2024-07-23 10:14:58.817122941 +0000
+++ b/package/Config.in	2024-07-23 10:09:37.299852762 +0000
@@ -2312,6 +2312,7 @@
 	source "package/gsettings-desktop-schemas/Config.in"
 	source "package/haveged/Config.in"
 	source "package/linux-syscall-support/Config.in"
+	source "package/min_prog_trace/Config.in"
 	source "package/mobile-broadband-provider-info/Config.in"
 	source "package/netdata/Config.in"
 	source "package/proj/Config.in"
